torch/distributed/device_mesh.py

```diff
@@ -338,6 +339,7 @@ def __eq__(self, other: object) -> bool:
                 and self._flatten_mesh_list == other._flatten_mesh_list
             )
 
+        @lru_cache(maxsize=None)
```

Comment:
hmmm iirc it's not safe to use lru cache on a class instance in python.

I think if you want to still use lru_cache, you could make lru_cache only live in the `MeshEnv` constructor instead, see https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/sharding_prop.py#L44

Reply:
nit: LRU cache I think stores a strong reference than a weak reference. Probably not a big deal in this case but just in case matters (like when you tear down all device meshes they wouldn't get GC'ed until you clear the cache).

Reply:
Also how does the LRU cache play with https://github.com/pytorch/pytorch/blob/8e74db9b1fc96123a28f9020664887df0fc63a0b/torch/distributed/device_mesh.py#L220C22-L220C42 ?